modules/infra/control/port.c

```diff
@@ -267,9 +267,11 @@ int iface_port_reconfig(
 		if (flags & GR_IFACE_F_UP) {
 			ret = rte_eth_dev_set_link_up(p->port_id);
 			iface->flags |= GR_IFACE_F_UP;
+			iface_event_notify(IFACE_EVENT_STATUS_UP, iface);
```

Comment:
This is just the administrative state. Don't you think we should only rely on the runtime state in link_event_cb?

Reply:
In that case, should we have 2 flags, for the physical link status and the administrative status ?
As of now, we don't have the admin state, so we can't put a vlan interface down for example.

Reply:
We do have two separate flag masks. iface->flags (for administrative status) and iface->state (for runtime status)

Reply:
VLAN interfaces runtime state is not properly managed though. It should follow their parent port runtime state.